cpp/src/parquet/properties.h

```diff
@@ -452,19 +452,39 @@ class PARQUET_EXPORT WriterProperties {
       return this->disable_statistics(path->ToDotString());
     }
 
-    /// Enable integer type to annotate decimal type as below:
-    ///   int32: 1 <= precision <= 9
-    ///   int64: 10 <= precision <= 18
-    /// Default disabled.
-    Builder* enable_integer_annotate_decimal() {
-      integer_annotate_decimal_ = true;
+    /// Enable decimal logical type with 1 <= precision <= 18 to be stored as
+    /// integer physical type.
+    ///
+    /// According to the specs, DECIMAL can be used to annotate the following types:
+    /// - int32: for 1 <= precision <= 9.
+    /// - int64: for 1 <= precision <= 18; precision < 10 will produce a warning.
+    /// - fixed_len_byte_array: precision is limited by the array size.
+    ///   Length n can store <= floor(log_10(2^(8*n - 1) - 1)) base-10 digits.
+    /// - binary: precision is not limited, but is required. precision is not limited,
+    ///   but is required. The minimum number of bytes to store the unscaled value
+    ///   should be used.
+    ///
+    /// By default, this is DISABLED and all decimal types annotate fixed_len_byte_array.
+    ///
+    /// When enabled, the C++ writer will use following physical types to store decimals:
+    /// - int32: for 1 <= precision <= 9.
+    /// - int64: for 10 <= precision <= 18.
+    /// - fixed_len_byte_array: for precision > 18.
+    ///
+    /// As a consequence, decimal columns stored in integer types are more compact
+    /// but in a risk that the parquet file may not be readable by previous Arrow C++
+    /// versions or other implementations.
```

Comment:
```suggestion
    /// As a consequence, decimal columns stored in integer types are more compact.
```

I'm not sure it's worth saying that older versions don't support them. In Arrow C++, read support for these was added at the same time for the other physical types for decimals ([original pr](https://github.com/apache/parquet-cpp/pull/403)). And that was a long time ago, before version 0.10.0. I'm sure it was similarly well supported in parquet-mr.

Reply:
FYI @pitrou

Reply:
I have checked that both apache/parquet-mr and apache/impala support reading this long ago. It seems that the native parquet reader in the velox does not handle this yet. I'm not familiar with other implementations but there may be some do not support it.

IMHO, we cannot take care of all other implementations. There is no risk to the parquet-cpp itself because the reader support is already in and it is well documented by the parquet specs. So I have removed the comment about the risk. @wjones127 @pitrou

Reply:
Thanks for confirming that.